inc/functions/files.php

```diff
@@ -979,11 +979,14 @@ function rocket_clean_cache_dir() {
 	*/
 	do_action( 'before_rocket_clean_cache_dir' ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
-	// Delete all caching files.
-	$dirs = glob( WP_ROCKET_CACHE_PATH . '*', GLOB_NOSORT );
-	if ( $dirs ) {
-		foreach ( $dirs as $dir ) {
-			rocket_rrmdir( $dir );
+	$filesystem = rocket_direct_filesystem();
+
+	// Delete all caching files/folders.
+	foreach ( _rocket_get_cache_dirs( '.' ) as $path ) {
+		if ( ! $filesystem->is_dir( $path ) ) {
+			$filesystem->delete( $path );
+		}else {
+			rocket_rrmdir( $path );
```

Comment:
```suggestion
		if ( ! $filesystem->is_dir( $path ) ) {
			$filesystem->delete( $path );
			continue;
		}

		rocket_rrmdir( $path );
```

Reply:
Really inside loops I don't like using continue and break as I think they are `hidden traps` and makes code not readable, what do you think?

Reply:
The same can be said of `else`, as some think `else` blocks are evil.

A good rule of thumb is this: 

- If no other processing should happen after the `if` block, then bail out: `return`, `continue`, or `break`.

Why?  Nothing else should happen after this `if` is done. Therefore, adding a bail out makes it very clear that the work is done.

Reply:
Great, nice Idea.
Thanks @Tabrisrp @hellofromtonya .